src/pages/instances/InstanceSnapshots.tsx

```diff
@@ -10,34 +10,63 @@ import {
 } from "@canonical/react-components";
 import { isoTimeToString } from "util/helpers";
 import { LxdInstance } from "types/instance";
-import CreateSnapshotForm from "pages/instances/actions/snapshots/CreateSnapshotForm";
 import NotificationRowLegacy from "components/NotificationRowLegacy";
-import SnapshotActions from "./actions/snapshots/SnapshotActions";
+import InstanceSnapshotActions from "./actions/snapshots/InstanceSnapshotActions";
 import useEventListener from "@use-it/event-listener";
 import Pagination from "components/Pagination";
 import { usePagination } from "util/pagination";
 import ItemName from "components/ItemName";
 import SelectableMainTable from "components/SelectableMainTable";
-import SnapshotBulkDelete from "pages/instances/actions/snapshots/SnapshotBulkDelete";
-import ConfigureSnapshotsBtn from "pages/instances/actions/snapshots/ConfigureSnapshotsBtn";
+import InstanceSnapshotBulkDelete from "pages/instances/actions/snapshots/InstanceSnapshotBulkDelete";
 import Loader from "components/Loader";
 import { useProject } from "context/project";
 import ScrollableTable from "components/ScrollableTable";
 import SelectedTableNotification from "components/SelectedTableNotification";
 import { useDocs } from "context/useDocs";
+import { useCreateInstanceSnapshot } from "./actions/snapshots/useInstanceSnapshot";
+import { isInstanceStateful } from "util/snapshots";
+import SnapshotForm from "components/forms/SnapshotForm";
+import PortalModalBtn from "components/PortalModalBtn";
+import InstanceConfigureSnapshotModal from "./actions/snapshots/InstanceConfigureSnapshotModal";
 
 const collapsedViewMaxWidth = 1250;
 export const figureCollapsedScreen = (): boolean =>
   window.innerWidth <= collapsedViewMaxWidth;
 
+interface SnapshotFormProps {
+  close: () => void;
+  instance: LxdInstance;
+  onSuccess: (message: ReactNode) => void;
+}
+
+const CreateInstanceSnapshotForm: FC<SnapshotFormProps> = (props) => {
```

Comment:
We should only have one component per file, where the filename matches the component name.